Fixed panic in m0_obj_init() when struct m0_obj_attr has some fields …

AbhishekSahaSeagate opened the pull request:
…populated

M0_PRE(M0_IS0(obj)) triggered panic when m0_obj::ob_attr:oa_obj_size
was set before calling m0_obj_init(). Removed this check to fix this.

Added an Unit Test to test setting of optimal layout id during
m0_entity_create().

Renamed the field from m0_obj::ob_attr:oa_obj_size to
m0_obj::ob_attr::oa_buf_size to be more complaint with motr design.

Signed-off-by: Abhishek Saha <abhishek.saha@seagate.com>

AbhishekSahaSeagate commented:
This PR is a fix for #713.

huanghua78 commented on motr/ut/obj.c:
```diff
@@ -658,6 +658,35 @@ static void ut_test_entity_namei_op_delete(void)
+	rc = m0_entity_create(NULL, &obj.ob_entity, &ops[0]);
+	M0_UT_ASSERT(obj.ob_attr.oa_layout_id > 0 &&
+			obj.ob_attr.oa_layout_id < 15);
+	M0_UT_ASSERT(rc = -ENOENT);
```
This is a bug.
```
M0_UT_ASSERT(rc == -ENOENT);
```
?

AbhishekSahaSeagate replied:
Ah, seems like I copied this bug along with the UT below this, I'll fix this and other UTs affected by this.

AbhishekSahaSeagate replied:
will have to change the check to 
```
M0_UT_ASSERT(rc == -EINVAL)
```
as `m0__obj_layout_instance_build()` returns -EINVAL.
```
        layout = m0_layout_find(&cinst->m0c_reqh.rh_ldom, layout_id);
        if (layout == NULL) {
                rc = -EINVAL;
                goto out;
        }
```

huanghua78 replied:
OK

AbhishekSahaSeagate commented:
@huanghua78 @andriytk @sg-shankar ,
The patch for https://github.com/Seagate/cortx-motr/pull/713 had an issue that, when m0_obj::ob_attr::oa_obj_size was set before calling m0_obj_init(), that caused this check to fail
```
        M0_PRE(parent != NULL);
        M0_PRE(id != NULL);
        M0_PRE(entity_id_is_valid(id));
-       M0_PRE(M0_IS0(obj));
        M0_PRE(layout_id < m0_lid_to_unit_map_nr);
```
so removed the check for now. If this is not a good approach, please suggest something.
 The setting of oa_obj_size before m0_obj_init() was chosen because if the oa_layout_id is set to 0 and oa_obj_size was not set before m0_entity_create(), this would cause invalid layout ID to be passed for entity creation.

andriytk commented:
@AbhishekSahaSeagate as mentioned at https://github.com/Seagate/cortx-motr/pull/713#discussion_r646547068, I'd suggest introducing a new function which would calculate the optimal unit size (layout id) based on the object size and then pass this layout id to obj_init(). And get rid of `oa_obj_size` which is very confusing.

andriytk reviewed (changes requested):
@AbhishekSahaSeagate as mentioned at https://github.com/Seagate/cortx-motr/pull/713#discussion_r646547068, I'd suggest introducing a new function which would calculate the optimal unit size (layout id) based on the object size, and then pass this layout id to obj_init().

And get rid of `oa_obj_size` which is very confusing.

Another (maybe even better and cleaner) approach would be to change obj_init() API and pass obj size argument to it instead of the layout id. But this might require a lot of changes (including 3rd party components which rely on the current API), so it's probably too late for this, I'm afraid. A new function will not break the existing API.

AbhishekSahaSeagate commented:
@andriytk, my initial implementation for this feature also involved creating a public API for S3 to call, but I changed it to match the requirements present in this doc: [document](https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/103055415/Fault+Tolerance+Handling+Analysis)

Perhaps @sg-shankar can shed more context on this if needed.

Is there any other way we can deal with the removal of this assert or is it fine if we removed it?
```
M0_PRE(M0_IS0(obj));
```
I don't see this requirement in the documentation of `m0_obj_init()` API as well
```
 * @pre obj != NULL
 * @pre parent != NULL
 * @pre id != NULL && m0_uint128_cmp(&M0_ID_APP, id) < 0
 */
void m0_obj_init(struct m0_obj           *obj,
                 struct m0_realm         *parent,
                 const struct m0_uint128 *id,
                 uint64_t                 layout_id);
```

andriytk commented:
> but I changed it to match the requirements present in this doc

I suppose you mean this one:

> 4. For Create object call, motr will provide pool version and layout ID for all scenario.

This requirement can be successfully fulfilled with the new function, which will return the optimal layout id based on the object size and the pool where the object is to be created.

The semantics of obj_init() function will also be preserved. User will not need to initialise some private object fields besides obj_init() before calling it.

huanghua78 commented:
> @andriytk, my initial implementation for this feature also involved creating a public API for S3 to call, but I changed it to match the requirements present in this doc: [document](https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/103055415/Fault+Tolerance+Handling+Analysis)
> 
> Perhaps @sg-shankar can shed more context on this if needed.
> 
> Is there any other way we can deal with the removal of this assert or is it fine if we removed it?
> 
> ```
> M0_PRE(M0_IS0(obj));
> ```
> 
> I don't see this requirement in the documentation of `m0_obj_init()` API as well
> 
> ```
>  * @pre obj != NULL
>  * @pre parent != NULL
>  * @pre id != NULL && m0_uint128_cmp(&M0_ID_APP, id) < 0
>  */
> void m0_obj_init(struct m0_obj           *obj,
>                  struct m0_realm         *parent,
>                  const struct m0_uint128 *id,
>                  uint64_t                 layout_id);
> ```


We need to update the pre-conditions of this function.
The "obj->ob_attr" may contain some input data. 
But we still can check:
M0_PRE(M0_IS0(&obj->ob_entity));

andriytk commented:
Here is the new function I mean - https://github.com/Seagate/cortx-motr/pull/753. Users can use it to get the optimal layout id and pass it into obj_init().

nikitadanilov commented on motr/ut/obj.c:
```diff
@@ -679,7 +708,7 @@ static void ut_test_m0_entity_create(void)
 			   m0_client_layout_id(instance));
 
 	rc = m0_entity_create(NULL, &obj.ob_entity, &ops[0]);
-	M0_UT_ASSERT(rc = -ENOENT);
```
Good catch!

huanghua78 replied:
Now, I know the reason why the -EINVAL is returned:
the layouts are not initialized. So we can not find layout with the specified lid.

huanghua78 replied:
So, please add comments to explain why this error is expected.

with the following patch,
```
diff --git a/motr/obj.c b/motr/obj.c
index 9ec61cb2..b4158b48 100644
--- a/motr/obj.c
+++ b/motr/obj.c
@@ -411,7 +411,7 @@ m0__obj_layout_instance_build(struct m0_client *cinst,
         */
        layout = m0_layout_find(&cinst->m0c_reqh.rh_ldom, layout_id);
        if (layout == NULL) {
-               rc = -EINVAL;
+               rc = M0_ERR(-EINVAL);
                goto out;
        }
 
@@ -527,15 +527,17 @@ static int obj_namei_op_init(struct m0_entity *entity,
                                        m0__obj_layout_id_get(oo));
        rc = m0__obj_layout_instance_build(cinst, lid,
                                           &oo->oo_fid, &linst);
-       if (rc != 0)
+       if (rc != 0) {
+               rc = M0_ERR(rc);
                goto error;
+       }
        oo->oo_layout_instance = linst;
 
 #ifdef CLIENT_FOR_M0T1FS
        /* Set the object's parent's fid. */
        if (!m0_fid_is_set(&cinst->m0c_root_fid) ||
            !m0_fid_is_valid(&cinst->m0c_root_fid)) {
-               rc = -EINVAL;
+               rc = M0_ERR(-EINVAL);
                goto error;
        }
        oo->oo_pfid = cinst->m0c_root_fid;
@@ -543,8 +545,10 @@ static int obj_namei_op_init(struct m0_entity *entity,
        /* Generate a valid oo_name. */
        obj_name = m0_alloc(M0_OBJ_NAME_MAX_LEN);
        rc = obj_fid_make_name(obj_name, M0_OBJ_NAME_MAX_LEN, &oo->oo_fid);
-       if (rc != 0)
+       if (rc != 0) {
+               rc = M0_ERR(rc);
                goto error;
+       }
        m0_buf_init(&oo->oo_name, obj_name, strlen(obj_name));
 #endif
        M0_ASSERT(rc == 0);

```
and when you run:
./scripts/m0 run-ut -t obj-ut:"m0_entity_create\(object\)" -e ERROR
You will see the error.

AbhishekSahaSeagate replied:
Done.

huanghua78 commented on motr/ut/obj.c:
```diff
@@ -658,6 +658,35 @@ static void ut_test_entity_namei_op_delete(void)
+	m0_obj_init(&obj, &realm, &ent.en_id, 0);
+
+	rc = m0_entity_create(NULL, &obj.ob_entity, &ops[0]);
+	M0_UT_ASSERT(obj.ob_attr.oa_layout_id > 0 &&
+			obj.ob_attr.oa_layout_id < 15);
```
Should we move this M0_UT_ASSERT() one line above to Line #680?
Because this is to check the result of m0_obj_init().

AbhishekSahaSeagate replied:
I don't think that would work as the optimal lid is assigned during `m0_entity_create()` in `obj_op_obj_init()`
```
        if (OP_OBJ2CODE(oo) == M0_EO_CREATE) {
                rc = m0__obj_pool_version_get(obj, &pv);
                if (rc != 0)
                        return M0_ERR(rc);
                oo->oo_pver = pv->pv_id;

                ldom = &cinst->m0c_reqh.rh_ldom;
                obj_optimal_lid_set(obj, ldom);
```
So we need to check after `m0_entity_create()` is complete.

huanghua78 replied:
OK

huanghua78 commented on motr/ut/obj.c:
```diff
@@ -658,6 +658,35 @@ static void ut_test_entity_namei_op_delete(void)
+	rc = m0_entity_create(NULL, &obj.ob_entity, &ops[0]);
+	M0_UT_ASSERT(obj.ob_attr.oa_layout_id > 0 &&
+			obj.ob_attr.oa_layout_id < 15);
+	M0_UT_ASSERT(rc == -EINVAL);
```
Why the result should be -EINVAL?
What are missing for this function call?

mehjoshi reviewed (approved):
Looks good to me.

cortx-admin commented:
<h1>Jenkins CI Result : <a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523/>Motr#523</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>1</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/02motr-single-node/51kem target=_blank>02motr-single-node/51kem</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>5</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/01motr-single-node/37protocol target=_blank>01motr-single-node/37protocol</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/04motr-single-node/31motr-sys-kvs-kernel-test target=_blank>04motr-single-node/31motr-sys-kvs-kernel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/07multi-client-tests target=_blank>05motr-single-node/07multi-client-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/12sns-cc-io-tests target=_blank>05motr-single-node/12sns-cc-io-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/39mount-fail target=_blank>05motr-single-node/39mount-fail</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>58</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/01motr-single-node/00userspace-tests target=_blank>01motr-single-node/00userspace-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/01motr-single-node/01kernel-tests target=_blank>01motr-single-node/01kernel-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/01motr-single-node/03initscript-tests target=_blank>01motr-single-node/03initscript-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/01motr-single-node/04net-tests target=_blank>01motr-single-node/04net-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/01motr-single-node/05rpcping-tests target=_blank>01motr-single-node/05rpcping-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/01motr-single-node/06console-tests target=_blank>01motr-single-node/06console-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/01motr-single-node/28motr-sys-kvs-test target=_blank>01motr-single-node/28motr-sys-kvs-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/01motr-single-node/29confgen-test target=_blank>01motr-single-node/29confgen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/01motr-single-node/30hagen-test target=_blank>01motr-single-node/30hagen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/01motr-single-node/35m0singlenode target=_blank>01motr-single-node/35m0singlenode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/01motr-single-node/46m0crate target=_blank>01motr-single-node/46m0crate</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/01motr-single-node/52singlenode-sanity target=_blank>01motr-single-node/52singlenode-sanity</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/02motr-single-node/02system-tests target=_blank>02motr-single-node/02system-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/02motr-single-node/08degraded-mode-tests target=_blank>02motr-single-node/08degraded-mode-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/02motr-single-node/09poolmach-tests target=_blank>02motr-single-node/09poolmach-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/02motr-single-node/20pool-version-test target=_blank>02motr-single-node/20pool-version-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/02motr-single-node/21spiel-test target=_blank>02motr-single-node/21spiel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/02motr-single-node/22rpc-cancel-test target=_blank>02motr-single-node/22rpc-cancel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/02motr-single-node/23m0d-signal-test target=_blank>02motr-single-node/23m0d-signal-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/02motr-single-node/24m0d-fsync-test target=_blank>02motr-single-node/24m0d-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/02motr-single-node/41motr-conf-update target=_blank>02motr-single-node/41motr-conf-update</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/02motr-single-node/47motr-tests-user-kernel target=_blank>02motr-single-node/47motr-tests-user-kernel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/02motr-single-node/53clusterusage-alert target=_blank>02motr-single-node/53clusterusage-alert</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/03motr-single-node/16conf-st-test target=_blank>03motr-single-node/16conf-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/03motr-single-node/17sss-st-test target=_blank>03motr-single-node/17sss-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/03motr-single-node/36spare-reservation target=_blank>03motr-single-node/36spare-reservation</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/03motr-single-node/54sns-repair-motr-1f target=_blank>03motr-single-node/54sns-repair-motr-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/03motr-single-node/55sns-repair-motr-1n-1f target=_blank>03motr-single-node/55sns-repair-motr-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/03motr-single-node/56sns-repair-motr-mf target=_blank>03motr-single-node/56sns-repair-motr-mf</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/03motr-single-node/57sns-repair-motr-1k-1f target=_blank>03motr-single-node/57sns-repair-motr-1k-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/03motr-single-node/58sns-repair-motr-ios-fail target=_blank>03motr-single-node/58sns-repair-motr-ios-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/03motr-single-node/59sns-repair-motr-abort target=_blank>03motr-single-node/59sns-repair-motr-abort</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/03motr-single-node/60sns-repair-motr-abort-quiesce target=_blank>03motr-single-node/60sns-repair-motr-abort-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/03motr-single-node/61spiel-multi-confd target=_blank>03motr-single-node/61spiel-multi-confd</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/04motr-single-node/19sns-abort-test target=_blank>04motr-single-node/19sns-abort-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/04motr-single-node/25spiel-sns-repair-test target=_blank>04motr-single-node/25spiel-sns-repair-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/04motr-single-node/26spiel-sns-repair-quiesce-test target=_blank>04motr-single-node/26spiel-sns-repair-quiesce-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/04motr-single-node/27sns-repair-io-fail-test target=_blank>04motr-single-node/27sns-repair-io-fail-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/04motr-single-node/32m0t1fs-rconfc-fail-test target=_blank>04motr-single-node/32m0t1fs-rconfc-fail-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/04motr-single-node/33motr-st target=_blank>04motr-single-node/33motr-st</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/04motr-single-node/34sns-repair-1n-1f target=_blank>04motr-single-node/34sns-repair-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/04motr-single-node/48motr-raid0-io target=_blank>04motr-single-node/48motr-raid0-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/04motr-single-node/49motr-rpc-cancel target=_blank>04motr-single-node/49motr-rpc-cancel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/04motr-single-node/50motr-rm-lock-cc-io target=_blank>04motr-single-node/50motr-rm-lock-cc-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/04motr-single-node/51motr-rmw target=_blank>04motr-single-node/51motr-rmw</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/10sns-single-tests target=_blank>05motr-single-node/10sns-single-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/11sns-multi-tests target=_blank>05motr-single-node/11sns-multi-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/13sns-repair-quiesce target=_blank>05motr-single-node/13sns-repair-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/14m0t1fs-fsync-test target=_blank>05motr-single-node/14m0t1fs-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/15m0t1fs-fwait-test target=_blank>05motr-single-node/15m0t1fs-fwait-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/18m0mount-test target=_blank>05motr-single-node/18m0mount-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/38sns-abort-quiesce-test target=_blank>05motr-single-node/38sns-abort-quiesce-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/40motr-dgmode target=_blank>05motr-single-node/40motr-dgmode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/42motr-utils target=_blank>05motr-single-node/42motr-utils</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/43motr-sync-replication target=_blank>05motr-single-node/43motr-sync-replication</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/44motr-sns-repair target=_blank>05motr-single-node/44motr-sns-repair</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/05motr-single-node/45motr-sns-repair-N-1 target=_blank>05motr-single-node/45motr-sns-repair-N-1</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit/motr-single-node/62dix-repair-st-lookup-insert target=_blank>motr-single-node/62dix-repair-st-lookup-insert</a></p></details></td></tr><tr><td>Total</td><td>64</td><td><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/523//testReport/junit>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

huanghua78 commented:
Hello @AbhishekSahaSeagate , Does this PR has a related Jira ticket?
If yes, please update the title of this PR to have the Jira ticket.
